addons/wiki/migrations/0009_migrate_node_wiki_pages.py

```diff
@@ -0,0 +1,195 @@
+                    guid.save()
+                    nwp = guid.referent
+                move_comment_target(Guid.load(wp._id), wp, nwp)
+                update_comments_viewed_timestamp(node, wp, nwp)
+    progress_bar.finish()
+    WikiVersion.objects.all().delete()
+    WikiPage.objects.all().delete()
+    logger.info('NodeWikiPages restored and WikiVersions and WikiPages removed.')
+
+def move_comment_target(current_guid, current_target, desired_target):
+    """Move the comment's target from the current target to the desired target"""
+    desired_target_guid = Guid.load(desired_target._id)
+    if Comment.objects.filter(Q(root_target_id=current_guid) | Q(target=current_guid)).exists():
+        Comment.objects.filter(root_target=current_guid).update(root_target=desired_target_guid)
+        Comment.objects.filter(target=current_guid).update(target=desired_target_guid)
+    return
+
+def update_comments_viewed_timestamp(node, current_wiki_object, desired_wiki_object):
+    """Replace the current_wiki_object keys in the comments_viewed_timestamp dict with the desired wiki_object_id """
+    contributors_pending_save = []
+    for contrib in node.contributors.exclude(comments_viewed_timestamp={}):
+        if contrib.comments_viewed_timestamp.get(current_wiki_object._id, None):
+            timestamp = contrib.comments_viewed_timestamp[current_wiki_object._id]
+            contrib.comments_viewed_timestamp[desired_wiki_object._id] = timestamp
+            del contrib.comments_viewed_timestamp[current_wiki_object._id]
+            contributors_pending_save.append(contrib)
+    bulk_update(contributors_pending_save, batch_size=1000)
+    return
+
+def migrate_guid_referent(guid, desired_referent, content_type_id):
+    """
+    Point the guid towards the desired_referent.
+    Pointing the NodeWikiPage guid towards the WikiPage will still allow links to work.
+    """
+    guid.content_type_id = content_type_id
+    guid.object_id = desired_referent.id
+    return guid
+
+def create_wiki_page(node, node_wiki, page_name):
+    wp = WikiPage(
+        page_name=page_name,
+        user_id=node_wiki.user_id,
+        node=node,
+        created=node_wiki.date,
+        modified=node_wiki.modified,
+    )
+    wp.update_modified = False
+    return wp
+
+def create_wiki_version(node_wiki, wiki_page):
+    wv = WikiVersion(
+        wiki_page=wiki_page,
+        user_id=node_wiki.user_id,
+        created=node_wiki.date,
+        modified=node_wiki.modified,
+        content=node_wiki.content,
+        identifier=node_wiki.version,
+    )
+    wv.update_modified = False
+    return wv
+
+def create_guids():
+    content_type = ContentType.objects.get_for_model(WikiPage)
+    progress_bar = progressbar.ProgressBar(maxval=WikiPage.objects.count()).start()
+    logger.info('Creating new guids for all WikiPages:')
+    for i, wp in enumerate(WikiPage.objects.all(), 1):
```

Comment:
Since we only need the `id` column, you can do `WikiPage.objects.values_list('id', flat=True)` here.